x-pack/plugins/ml/server/routes/system.ts

```diff
@@ -277,6 +277,52 @@ export function systemRoutes(
             return acc;
           }, {} as Record<string, { exists: boolean }>);
 
+          return response.ok({
+            body: result,
+          });
+        } catch (error) {
+          return response.customError(wrapError(error));
+        }
+      })
+    );
+
+  /**
+   * @apiGroup SystemRoutes
+   *
+   * @api {post} /internal/ml/reindex_with_pipeline ES reindex wrapper to reindex with pipeline
+   * @apiName MlReindexWithPipeline
+   */
+  router.versioned
+    .post({
+      path: `${ML_INTERNAL_BASE_PATH}/reindex_with_pipeline`,
+      access: 'internal',
+      options: {
+        tags: ['access:ml:canGetMlInfo'],
```

Comment:
`canGetMlInfo` a viewer capability. 
Although the actual call to es will fail with the wrong user permissions, I still think we should have an admin capability check on this route.
`canCreateTrainedModels` is probably the most applicable.

Reply:
Updated in 78af285e55c9eef7203b9d88ee617ea306cf460a